purchases/src/main/kotlin/com/revenuecat/purchases/blockstore/BlockstoreHelper.kt

```diff
@@ -0,0 +1,140 @@
+package com.revenuecat.purchases.blockstore
+
+import android.content.Context
+import com.google.android.gms.auth.blockstore.Blockstore
+import com.google.android.gms.auth.blockstore.BlockstoreClient
+import com.google.android.gms.auth.blockstore.DeleteBytesRequest
+import com.google.android.gms.auth.blockstore.RetrieveBytesRequest
+import com.google.android.gms.auth.blockstore.RetrieveBytesResponse.BlockstoreData
+import com.google.android.gms.auth.blockstore.StoreBytesData
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.PurchasesException
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.identity.IdentityManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+internal class BlockstoreHelper(
+    applicationContext: Context,
+    private val identityManager: IdentityManager,
+    private val blockstoreClient: BlockstoreClient = Blockstore.getClient(applicationContext),
+    private val ioScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.IO),
+    private val mainScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main),
+) {
+
+    private companion object {
+        const val BLOCKSTORE_USER_ID_KEY = "revenuecat_blockstore_user_id"
```

Comment:
I'd suggest using a fully qualified package name, and the `blockstore` part is redundant imo. Something like: 
```suggestion
        const val BLOCKSTORE_USER_ID_KEY = "com.revenuecat.purchases.app_user_id"
```